Broker's spec.delivery.retry being nil causes nil pointer derefrence

**Describe the bug**
https://github.com/google/knative-gcp/blob/87b9a08562d4019eb35522b616b67a0fbe0b9ed0/pkg/reconciler/trigger/trigger.go#L319
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1ae887a]

goroutine 790 [running]:
github.com/google/knative-gcp/pkg/reconciler/trigger.getPubsubDeadLetterPolicy(0xc00058c726, 0x9, 0xc00063a6c0, 0xc000cb7d10)
        github.com/google/knative-gcp/pkg/reconciler/trigger/trigger.go:319 +0x11a
github.com/google/knative-gcp/pkg/reconciler/trigger.(*Reconciler).reconcileRetryTopicAndSubscription(0xc000b792c0, 0x2399520, 0xc000f4e000, 0xc001bdd680, 0xc00063a6c0, 0x0, 0x0)
        github.com/google/knative-gcp/pkg/reconciler/trigger/trigger.go:267 +0xc5d
github.com/google/knative-gcp/pkg/reconciler/trigger.(*Reconciler).reconcile(0xc000b792c0, 0x2399520, 0xc000f4e000, 0xc001bdd680, 0xc000a5d080, 0x0, 0x28)
        github.com/google/knative-gcp/pkg/reconciler/trigger/trigger.go:158 +0x1d7
github.com/google/knative-gcp/pkg/reconciler/trigger.(*Reconciler).ReconcileKind(0xc000b792c0, 0x2399520, 0xc000f4e000, 0xc001bdd680, 0x23652a0, 0xc001bdd7c0)
        github.com/google/knative-gcp/pkg/reconciler/trigger/trigger.go:143 +0x625
github.com/google/knative-gcp/pkg/client/injection/reconciler/broker/v1beta1/trigger.(*reconcilerImpl).Reconcile(0xc0008e28c0, 0x2399520, 0xc000efdfb0, 0xc001bf6180, 0x15, 0xc0014b0690, 0x2399520)
        github.com/google/knative-gcp/pkg/client/injection/reconciler/broker/v1beta1/trigger/reconciler.go:241 +0x1262
knative.dev/pkg/controller.(*Impl).processNextWorkItem(0xc000bad500, 0x5a64566234764100)
        knative.dev/pkg@v0.0.0-20210107022335-51c72e24c179/controller/controller.go:523 +0x6be
knative.dev/pkg/controller.(*Impl).RunContext.func3(0xc001b55384, 0xc000bad500)
        knative.dev/pkg@v0.0.0-20210107022335-51c72e24c179/controller/controller.go:461 +0x53
created by knative.dev/pkg/controller.(*Impl).RunContext
        knative.dev/pkg@v0.0.0-20210107022335-51c72e24c179/controller/controller.go:459 +0x1a5
```

 **Expected behavior**
No panics. Leave the field blank and let PubSub default its value.

 **To Reproduce**
With the default `config-br-delivery`, which does not specify a retry number, apply:

```
apiVersion: eventing.knative.dev/v1beta1
kind: Broker
metadata:
  name: test-broker
  annotations:
    "eventing.knative.dev/broker.class": "googlecloud"
spec:
  delivery:
    deadLetterSink:
      uri: pubsub://foo
```

See that the controller continuously panics.

--- a/pkg/reconciler/trigger/trigger.go
+++ b/pkg/reconciler/trigger/trigger.go
@@ -315,10 +315,13 @@ func getPubsubDeadLetterPolicy(projectID string, spec *eventingduckv1beta1.Deliv
 		return nil
 	}
 	// Translate to the pubsub dead letter policy format.
-	return &pubsub.DeadLetterPolicy{
-		MaxDeliveryAttempts: int(*spec.Retry),
-		DeadLetterTopic:     fmt.Sprintf("projects/%s/topics/%s", projectID, spec.DeadLetterSink.URI.Host),
+	dlp := &pubsub.DeadLetterPolicy{
+		DeadLetterTopic: fmt.Sprintf("projects/%s/topics/%s", projectID, spec.DeadLetterSink.URI.Host),
 	}
+	if spec.Retry != nil {
+		dlp.MaxDeliveryAttempts = int(*spec.Retry)
+	}
+	return dlp
 }
 
 func (r *Reconciler) deleteRetryTopicAndSubscription(ctx context.Context, trig *brokerv1beta1.Trigger) error {

--- a/pkg/reconciler/trigger/trigger_test.go
+++ b/pkg/reconciler/trigger/trigger_test.go
@@ -98,6 +98,16 @@ var (
 			},
 		},
 	}
+	brokerDeliverySpecWithoutRetry = &eventingduckv1beta1.DeliverySpec{
+		BackoffDelay:  &backoffDelay,
+		BackoffPolicy: &backoffPolicy,
+		DeadLetterSink: &duckv1.Destination{
+			URI: &apis.URL{
+				Scheme: "pubsub",
+				Host:   deadLetterTopicID,
+			},
+		},
+	}
 )
 
 func init() {
@@ -475,6 +485,73 @@ func TestAllCasesTrigger(t *testing.T) {
 						MaxDeliveryAttempts: 3,
 						DeadLetterTopic:     "projects/test-project-id/topics/test-dead-letter-topic-id",
 					}),
+			},
+		},
+		{
+			Name: "Check topic config and labels - broker without spec.delivery.retry",
+			Key:  testKey,
+			Objects: []runtime.Object{
+				NewBroker(brokerName, testNS,
+					WithBrokerClass(brokerv1beta1.BrokerClass),
+					WithInitBrokerConditions,
+					WithBrokerReady("url"),
+					WithBrokerDeliverySpec(brokerDeliverySpecWithoutRetry),
+					WithBrokerSetDefaults,
+				),
+				makeSubscriberAddressableAsUnstructured(),
+				NewTrigger(triggerName, testNS, brokerName,
+					WithTriggerUID(testUID),
+					WithTriggerSubscriberRef(subscriberGVK, subscriberName, testNS),
+					WithTriggerSetDefaults),
+			},
+			WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+				Object: NewTrigger(triggerName, testNS, brokerName,
+					WithTriggerUID(testUID),
+					WithTriggerSubscriberRef(subscriberGVK, subscriberName, testNS),
+					WithTriggerBrokerReady,
+					WithTriggerSubscriptionReady,
+					WithTriggerTopicReady,
+					WithTriggerDependencyReady,
+					WithTriggerSubscriberResolvedSucceeded,
+					WithTriggerStatusSubscriberURI(subscriberURI),
+					WithTriggerSetDefaults,
+				),
+			}},
+			WantEvents: []string{
+				triggerFinalizerUpdatedEvent,
+				topicCreatedEvent,
+				subscriptionCreatedEvent,
+				triggerReconciledEvent,
+			},
+			WantPatches: []clientgotesting.PatchActionImpl{
+				patchFinalizers(testNS, triggerName, finalizerName),
+			},
+			OtherTestData: map[string]interface{}{
+				"pre": []PubsubAction{
+					Topic("test-dead-letter-topic-id"),
+				},
+				"dataResidencyConfigMap": NewDataresidencyConfigMapFromRegions([]string{"us-east1"}),
+			},
+			PostConditions: []func(*testing.T, *TableRow){
+				OnlyTopics("cre-tgr_testnamespace_test-trigger_abc123", "test-dead-letter-topic-id"),
+				OnlySubscriptions("cre-tgr_testnamespace_test-trigger_abc123"),
+				SubscriptionHasRetryPolicy("cre-tgr_testnamespace_test-trigger_abc123",
+					&pubsub.RetryPolicy{
+						MaximumBackoff: 5 * time.Second,
+						MinimumBackoff: 5 * time.Second,
+					}),
+				SubscriptionHasDeadLetterPolicy("cre-tgr_testnamespace_test-trigger_abc123",
+					&pubsub.DeadLetterPolicy{
+						DeadLetterTopic: "projects/test-project-id/topics/test-dead-letter-topic-id",
+					}),
+				TopicExistsWithConfig("cre-tgr_testnamespace_test-trigger_abc123", &pubsub.TopicConfig{
+					MessageStoragePolicy: pubsub.MessageStoragePolicy{
+						AllowedPersistenceRegions: []string{"us-east1"},
+					},
+					Labels: map[string]string{
+						"name": "test-trigger", "namespace": "testnamespace", "resource": "triggers",
+					},
+				}),
 			},
 		},
 		{

